test/integration/src/test/java/io/pravega/test/integration/RestoreBackUpDataRecoveryTest.java

```
@@ -444,6 +484,495 @@ public void testDurableDataLogFail() throws Exception {
+        int numOfSegments = configuration.getScalingPolicy().getMinNumSegments();
+        double delta = 1.0 / numOfSegments;
+        for (long segmentNumber = 0; segmentNumber < numOfSegments - 1; segmentNumber++) {
+            double rangeLow = segmentNumber * delta;
+            double rangeHigh = (segmentNumber + 1) * delta;
+            double rangeMid = (rangeHigh + rangeLow) / 2;
+
+            Map<Double, Double> map = new HashMap<>();
+            map.put(rangeLow, rangeMid);
+            map.put(rangeMid, rangeHigh);
+            controller.scaleStream(streamObj, Collections.singletonList(segmentNumber), map, executorService()).getFuture().join();
+        }
+    }
+
+    private EventStreamReader<String> createReader(ClientFactoryImpl clientFactory,
+                                                   ReaderGroupManager readerGroupManager, String scope, String stream,
+                                                   String readerGroupName, String readerName) {
+        readerGroupManager.createReaderGroup(readerGroupName,
+                ReaderGroupConfig
+                        .builder()
+                        .stream(Stream.of(scope, stream))
+                        .automaticCheckpointIntervalMillis(2000)
+                        .build());
+
+        return clientFactory.createReader(readerName,
+                readerGroupName,
+                new UTF8StringSerializer(),
+                ReaderConfig.builder().build());
+    }
+
+    private Position readNEvents(EventStreamReader<String> reader, int num) {
+        Position position = null;
+        for (int q = 0; q < num;) {
+            try {
+                EventRead<String> eventRead = reader.readNextEvent(SECONDS.toMillis(5000));
+                Assert.assertEquals("Event written and read back don't match", EVENT, eventRead.getEvent());
+                q++;
+                position = eventRead.getPosition();
+            } catch (Exception e) {
```

Isn't the exception bubbling up going to have the same effect as your `Assert.fail` (i.e., stop the test)? You are suppressing this exception so you will never know the type of exception or its stack trace. My recommendation is to bubble it up.

---

Letting them bubble up.